### ISSUE TITLE
hidden tm->dialog dependency during shutdown

### ISSUE THREAD

**Issue by 46labs** [OpenSIPS/opensips#730]
OpenSIPS 2.2, commit ab5cd61
```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007fa0585613ea in unref_dlg (dlg=0x7fa0609fcea8, cnt=1) at dlg_hash.c:791
791		d_entry = &(d_table->entries[dlg->h_entry]);
Traceback (most recent call last):
  File "/usr/share/gdb/auto-load/usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.19-gdb.py", line 63, in <module>
    from libstdcxx.v6.printers import register_libstdcxx_printers
ImportError: No module named 'libstdcxx'
(gdb) bt full
#0  0x00007fa0585613ea in unref_dlg (dlg=0x7fa0609fcea8, cnt=1) at dlg_hash.c:791
        d_entry = <optimized out>
        __FUNCTION__ = "unref_dlg"
#1  0x00007fa05c159b48 in empty_tmcb_list (head=head@entry=0x7fa06340abc0) at t_hooks.c:53
        cbp = 0x0
        cbp_tmp = 0x7fa05ed127a0
#2  0x00007fa05c13bed5 in free_cell (dead_cell=0x7fa06340ab50) at h_table.c:127
        b = <optimized out>
        i = <optimized out>
        rpl = <optimized out>
        tt = <optimized out>
        foo = <optimized out>
        p = <optimized out>
#3  0x00007fa05c13cefa in free_hash_table () at h_table.c:375
No locals.
#4  0x00007fa05c153a9b in tm_shutdown () at t_funcs.c:91
        __FUNCTION__ = "tm_shutdown"
#5  0x00000000004c350a in destroy_modules () at sr_module.c:527
No locals.
#6  0x000000000043fd1a in cleanup (show_status=show_status@entry=1) at main.c:308
        __FUNCTION__ = "cleanup"
#7  0x0000000000440768 in handle_sigs () at main.c:503
        chld = 0
        chld_status = 139
        overall_status = 139
        i = <optimized out>
        do_exit = <optimized out>
        __FUNCTION__ = "handle_sigs"
#8  0x00000000004192ea in main_loop () at main.c:746
        startup_done = 0x0
        chd_rank = 232
#9  main (argc=<optimized out>, argv=<optimized out>) at main.c:1250
        cfg_log_stderr = <optimized out>
---Type <return> to continue, or q <return> to quit---
        cfg_stream = <optimized out>
        c = <optimized out>
        r = <optimized out>
        tmp = 0x7ffe0bca6e8b ""
        tmp_len = <optimized out>
        port = <optimized out>
        proto = <optimized out>
        options = 0x5c0470 "f:cCm:M:b:l:n:N:rRvdDFETSVhw:t:u:g:P:G:W:o:"
        seed = 1918274510
        __FUNCTION__ = "main"
```

Logs:
```
Dec 17 20:10:57 r5-c1-n3-lb /usr/local/sbin/opensips[11474]: CRITICAL:dialog:unref_dlg: bogus ref -1 with cnt 1 for dlg 0x7fa06068db68 [465:634477760] with clid '4C793887-56736B24000BE3F2-05A18700' and tags '226E8DB1-56736B24000BE3F0-05A18700' 'SD0rl6899-yS2SDvSFgBrXB'
```

**Comment by liviuchircu** [comment]
Updated issue name and improved its content. Trevor also configured the OS so that future corefiles will not overwrite each other. Waiting for a proper backtrace.

**Comment by 46labs** [comment]
Just dropping a comment to remind that this has to deal with start order of modules (until patch is applied) Cluster, then dialog, then TM resolves this crash.

**Comment by liviuchircu** [comment]
This has been fixed in March by @razvancrainea in 2600e002. Closing.
